packages/compass/src/app/auto-connect.spec.ts

```diff
@@ -1,20 +1,66 @@
 import { expect } from 'chai';
 import sinon from 'sinon';
 import { loadAutoConnectInfo } from './auto-connect';
-import { exportConnections } from '@mongodb-js/connection-storage';
-import type { ConnectionInfo } from '@mongodb-js/connection-storage';
+import {
+  ConnectionStorage,
+  type ExportConnectionOptions,
+} from '@mongodb-js/connection-storage/main';
+import type { ConnectionInfo } from '@mongodb-js/connection-storage/renderer';
 import { ipcRenderer } from 'hadron-ipc';
+import fs from 'fs/promises';
+import path from 'path';
+import os from 'os';
+
+const loadAutoConnectWithConnection = async (
+  connectPreferences: Record<string, unknown> = {},
+  connections: ConnectionInfo[] = [],
+  exportOptions: ExportConnectionOptions = {}
+) => {
+  const tmpDir = await fs.mkdtemp(
+    path.join(os.tmpdir(), 'connection-storage-tests')
+  );
+  try {
+    await fs.mkdir(path.join(tmpDir, 'Connections'));
+
+    ConnectionStorage['path'] = tmpDir;
+    await Promise.all(
+      connections.map((connectionInfo) =>
+        ConnectionStorage.save({ connectionInfo })
+      )
+    );
+
+    const fileContents = await ConnectionStorage.exportConnections({
+      options: exportOptions,
+    });
+
+    const fakeFs = { readFile: sinon.stub().resolves(fileContents) };
+    const deserializeConnections =
+      ConnectionStorage.deserializeConnections.bind(ConnectionStorage);
+
+    const fn = await loadAutoConnectInfo(
+      sinon.stub().resolves(connectPreferences),
+      fakeFs,
+      deserializeConnections
+    );
+    return fn;
+  } finally {
+    void fs.rmdir(tmpDir, { recursive: true });
+  }
+};
 
 describe('auto connection argument parsing', function () {
   let sandbox: sinon.SinonSandbox;
+  const initialKeytarEnvValue = process.env.COMPASS_E2E_DISABLE_KEYCHAIN_USAGE;
 
   beforeEach(function () {
     sandbox = sinon.createSandbox();
     sandbox.stub(ipcRenderer, 'call').resolves(true);
+    process.env.COMPASS_E2E_DISABLE_KEYCHAIN_USAGE = 'true';
   });
 
   afterEach(function () {
     sandbox.restore();
+    process.env.COMPASS_E2E_DISABLE_KEYCHAIN_USAGE = initialKeytarEnvValue;
```

Comment:
Note that

```js
> process.env.ENVVAR = process.env.ENVVAR
undefined
> process.env.ENVVAR 
'undefined'
```

so this does not fully reset `COMPASS_E2E_DISABLE_KEYCHAIN_USAGE` back to its original state here, probably doesn’t matter all that much because in the places where we check `COMPASS_E2E_DISABLE_KEYCHAIN_USAGE` we explicitly check for the string `'true'`, but that’s a Compass-specific choice and not very idiomatic, so I wouldn’t really rely on it:

```suggestion
    if (initialKeytarEnvValue)
      process.env.COMPASS_E2E_DISABLE_KEYCHAIN_USAGE = initialKeytarEnvValue;
    else 
      delete process.env.COMPASS_E2E_DISABLE_KEYCHAIN_USAGE;
```

which is more verbose but should be quite a bit safer

Reply:
thank you, and this is really how we should reset it.